### DISCUSSION TITLE
http://www.andrewheiss.com/blog/2018/07/30/disposable-supercomputer-future/

### DISCUSSION THREAD

**Discussion by andrewheiss** [andrewheiss/ath-quarto#32] [Blog comments]
Create a cheap, disposable supercomputer with R, DigitalOcean, and future | Andrew Heiss 
 http://www.andrewheiss.com/blog/2018/07/30/disposable-supercomputer-future/

**Comment by andrewheiss** [comment]
```header
Originally posted by César Arquero Cabral on 2019-04-30 11:28:37
````
Hi, I\'m trying to implement exactly this but I\'m stack in the
&quot;ssh_private_key_file \&lt;- &quot;/Users/andrew/.ssh/id_rsa&quot; part.\
This is the only file that I have within that dir:
&quot;ssh_private_key_file \&lt;- &quot;/home/cesarkero/.ssh/known_hosts&quot;\
When I execute the makeclusterpsock functions it appear this message:
WARNING: UNPROTECTED PRIVATE KEY FILE!\
I\'ve tryed to redirect the ssh_private_key_file to both private and
public ppk made with puttygen\...but nothing works.\
Do you have any advise ?

**Reply by andrewheiss** [reply]
```header
Originally posted by César Arquero Cabral on 2019-04-30 11:31:38
````
Here\'s is what I get:
[https://www.evernote.com/l/\...](https://www.evernote.com/l/AjloBwCnrzpJiq1YswnEA5eXlNPO3N3MQ6E/)

**Reply by andrewheiss** [reply]
```header
Originally posted on 2019-04-30 15:24:08
````
You need to adjust the permissions for \~/.ssh:
[https://www.howtogeek.com/1\...](https://www.howtogeek.com/168119/fixing-warning-unprotected-private-key-file-on-linux/ &quot;https://www.howtogeek.com/168119/fixing-warning-unprotected-private-key-file-on-linux/&quot;)

**Reply by andrewheiss** [reply]
```header
Originally posted by César Arquero Cabral on 2019-05-02 18:13:31
````
I have finally resolve it and now i can conect to those machines. How
could i do now a process that uses a huge csv (raster) data within the
main machine? Does the data need to be downloaded into the
workermachines ?

**Reply by andrewheiss** [reply]
```header
Originally posted on 2019-05-02 18:14:58
````
Yeah, the data has to be present on each of the machines

**Comment by andrewheiss** [comment]
```header
Originally posted by Paul John on 2019-11-29 17:30:43
````
How can i implement this if i have owned DigitalOcean Hosting from
Cloudways platform. My main application is hosted there but too much
space available to try this out

**Comment by Tadge-Analytics** [comment]
Is it possible to use the cores across the remote cluster as well as the cores of the local machine (all at the same time?) -great + very hepful blogpost btw 😁

**Comment by WihanZA** [comment]
Hi Andrew. Thanks for the very useful post. I was wondering: If each docklet is running Ubuntu (a POSIX system), would it not be more preferable to use a makeForkCluster()? Also: Could you please shed some light on your decision to use plan(cluster, workers = cl) as opposed to some other plan, such as those using tweak()? Please excuse my ignorance if I'm missing something obvious. Thanks again for the good work!

**Reply by andrewheiss** [reply]
Mostly because I wrote this in 2018 in the early days of the {future} package, before other functions like `tweak()` existed :)
